tsMuxer/blurayHelper.cpp

```diff
@@ -165,22 +166,29 @@ bool BlurayHelper::writeBluRayFiles(bool usedBlackPL, int mplsNum, int blankNum,
         delete file;
         return false;
     }
-
+    uint8_t* emptyCommand;
     if (m_dt == DT_BLURAY) {
         bdMovieObjectData[5] = bdIndexData[5] = '2';
         fileSize = 0x78;
     }
     else if (m_dt == UHD_BLURAY) {
         bdMovieObjectData[5] = bdIndexData[5] = '3';
-        fileSize = 0x78;
+        fileSize = 0x9C; // add 36 bytes for HDR data extension
+        bdIndexData[15] = 0x78; // start address of HDR data extension
+        emptyCommand = bdIndexData + 0x78;
+        memcpy(emptyCommand, "\x00\x00\x00\x20\x00\x00\x00\x18\x00\x00\x00\x01"
+                             "\x00\x03\x00\x01\x00\x00\x00\x18\x00\x00\x00\x0C"
+                             "\x00\x00\x00\x08\x51\x00\x00\x00\x00\x00\x00\x00", 36); // HDR data extension
+        bdIndexData[0x96] = (byte)HDR10_metadata[0]; // HDR flags

```

Comment:
`std::byte` is a C++17 feature, and this code rightly causes a compilation failure with Clang. `static_cast` to `uint8_t` instead.

Reply:
Looks like [GCC doesn't like this either](https://github.com/jcdr428/tsMuxer/commit/d8d9669973af16b4e6c64eb6b0d08686327ad099/checks#step:5:62).